File: src/test/java/com/bbcnews/automation/testutils/AppiumPortFactory.java
package com.bbcnews.automation.testutils;

import java.net.ServerSocket;

public class AppiumPortFactory {
    public static int create() throws Exception {
        ServerSocket socket = new ServerSocket(0);
        socket.setReuseAddress(true);
        int port = socket.getLocalPort();
        socket.close();
        return port;
    }

    public static void main(String[] args) throws Exception
    {
        AppiumPortFactory portFactory = new AppiumPortFactory();
        for (int i = 0; i < 10; i++)
            System.out.println(portFactory.create());
    }

}

File: src/test/java/com/bbcnews/automation/testutils/AppiumPortFactory.kt
package com.bbcnews.automation.testutils

import java.net.ServerSocket

object AppiumPortFactory {
    @Throws(Exception::class)
    fun create(): Int {
        val socket = ServerSocket(0)
        socket.reuseAddress = true
        val port = socket.localPort
        socket.close()
        return port
    }



}